watsononlinestore/watson_online_store.py

```diff
@@ -303,6 +312,19 @@ def get_discovery_response(self, input_text):
             query_options={'query': input_text, 'count': DISCOVERY_QUERY_COUNT}
         )
 
+        # Watson discovery assigns a confidence level to each result.
+        # Based on data mix, we can assign a minimum tolerance value in an
+        # attempt to filter out the "weakest" results.
+        if self.discovery_score_filter and 'results' in discovery_response:
+            filtered_results = []
+            for result in discovery_response['results']:
```

Comment:
i still think this can be written as:

  `filtered_results = [x for x in discovery_response['results'] if 'score' in x and x['score'] > self.discovery_score_filter`

probably need shorter variable names to make it happen